### PR TITLE
feat(cli): [STENCIL-12] Add Telemetry to Stencil CLI 

### PR DESCRIPTION
**Pull request by splitinfinities**
This is a chained PR, do not merge it!

Adds Telemetry options to the CLI. Try it out by running any of these commands in your linked project: 

`npx @stencil/core telemetry`
`npx @stencil/core telemetry on|off`

The debug flag will print the telemetry results, e.g:
`npx @stencil/core generate a-component --debug`

Satisfies ADR-1

### CODE REVIEW

File: src/cli/ionic-config.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { readJSON, writeJSON, mkdirp } from '@ionic/utils-fs';
```
Comment (ltm) [inline src/cli/ionic-config.ts:1]: I don't think you can import this without adding it to `dependencies`. Given our zero dependencies policy you may just need to copy the code from the ionic repo to the stencil repo.

**Comment by splitinfinities** [comment]
Closing to point at the STENCIL-33 branch

File: src/cli/telemetry/ipc.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+/**
+ * Receive and handle an IPC message.
+ *
+ * Assume minimal context and keep external dependencies to a minimum.
+ */
+export async function receive(msg: IPCMessage): Promise<void> {
+	debug('Received %O IPC message', msg.type);
+	console.log("WOOHOO")
```
Comment (ltm) [inline src/cli/telemetry/ipc.ts:48]:
```suggestion
```

File: src/cli/ionic-config.ts (hunk 1 above)
Comment (ltm) [reply src/cli/ionic-config.ts:1]: The same applies to `@ionic/utils-terminal` and `@ionic/utils-subprocess`.

**Pull request closed without merging**